PayPalNative/src/main/java/com/braintreepayments/api/PayPalNativeClient.java

```diff
@@ -50,33 +79,130 @@ public void onResult(Configuration configuration, Exception error) {
+        final String redirectUrl = getPayPalReturnUrl(activity);
+        Environment environment = Environment.SANDBOX;
+        if ("production".equalsIgnoreCase(configuration.getEnvironment())) {
+            environment = Environment.LIVE;
+        }
+
+        CheckoutConfig checkoutConfig = new CheckoutConfig(
+                activity.getApplication(),
+                clientId,
+                environment,
+                redirectUrl
+        );
+        PayPalCheckout.setConfig(checkoutConfig);
+        PayPalCheckout.start(
+                new CreateOrder() {
+                    @Override
+                    public void create(final CreateOrderActions createOrderActions) {
+                        createOrderActions.set(payPalResponse.getPairingId());
+                    }
+                },
+                new OnApprove() {
+                    @Override
+                    public void onApprove(final Approval approval) {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.approved");
+                        approved(approval, payPalResponse, callback);
+                    }
+                },
+                new OnCancel() {
+                    @Override
+                    public void onCancel() {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.client_cancel");
+                        callback.onResult(null, new BraintreeException("User Cancelled"));
+                    }
+                },
+                new OnError() {
+                    @Override
+                    public void onError(final ErrorInfo errorInfo) {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.error");
+                        callback.onResult(null, new BraintreeException(errorInfo.getReason()));
+                    }
+                }
+        );
+    }
+
+    //TODO: I copied all of this from PayPal client and adapted it. Is all of this necessary?
+    private void approved(final Approval approval, final PayPalResponse payPalResponse, final PayPalNativeFlowCallback callback) {
+        Uri deepLinkUri = Uri.parse(String.format(
+                "%s://onetouch/v1/success?paymentId=%s&token=%s&PayerID=%s",
+                braintreeClient.getReturnUrlScheme(),
+                approval.getData().getPaymentId(),
+                approval.getData().getOrderId(),
+                approval.getData().getPayerId()));
+        try {
+            JSONObject urlResponseData = parseUrlResponseData(deepLinkUri, payPalResponse.getSuccessUrl(), payPalResponse.getApprovalUrl(), "token");
+            PayPalAccount payPalAccount = new PayPalAccount();
+            payPalAccount.setClientMetadataId(payPalResponse.getClientMetadataId());
+            payPalAccount.setSource("paypal-browser"); // paypal native throws error
+            payPalAccount.setUrlResponseData(urlResponseData);
+            payPalAccount.setPaymentType("single-payment"); // is this the correct way?
```

Comment:
If I pass `paypal-native` to `setSource()`, `tokenizeREST` returns an error. Where can we configure that to accept `paypal-native` (or maybe something else?). Is also 'single-payment` the correct payment type?

Reply:
I believe these fields are outside of the SDK domain, as they are used for tokenizing the PayPal account. We should look into which are required/can be modified to align with this new flow.

Reply:
do you know where i can take a look at which are the required fields?